src/main/java/com/teragrep/rlp_03/SocketProcessor.java

```diff
@@ -432,7 +422,6 @@ private void processReadWriteClose(
             selectionKey.channel().close();
             selectionKey.cancel();
 
-            //this.socketMap.remove(socket);
```

Comment:
please uncomment the remove and try again

Reply:
Edited the PR